helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -678,52 +680,56 @@ private void syncFactoryState() {
   }
 
   void reset() {
-    LOG.info("Reset HelixTaskExecutor");
+    synchronized (_resetLock) {
```

Comment:
do you need to lock the whole method?? if so, why not add synchronized at method itself.

Reply:
Synchronized method is not used since I don't think this has to be mutual exclusive to other methods synchronized on the instance.
Do we need this synchronization at all? Maybe not, I'm still trying staging changes hoping to better improve the reset().

Reply:
synchronized method doesn't mean across different methods. only one call to this method. 
I agree that  you don't need synchronization. reset() method has 3 states: No one has invoked it. it is in progress and it has completed the job and nothing to reset. So if you model as appropriate status enum, you can work around where reset() is called only once in shutdown path. my 2cents

Reply:
Thanks for the comment.
I updated a bit to use synchronization and a boolean state, as I find in_progress state is unnecessary. Hope that makes sense.